live/liveview.go

```diff
@@ -757,6 +769,24 @@ type Event struct {
 	Data url.Values
 }
 
+// MarshalJSON implements json.Marshaler for Event
+func (e *Event) MarshalJSON() ([]byte, error) {
+	// if a key is multi-valued, send it as an array
```

Comment:
I know this is not our protocol to define, but this seems ambiguous to me. What if the single value is itself an array?

This protocol feels over-engineered to me. It should have been simpler and let gzip do all the "optimization" work.

Anyway, grumpy old man here signing off.

Reply:
honestly not sure of the protocol as there isn't a specification.

Reply:
Almost certainly a consequence of elixir pattern-matching: lists vs. single value means easier pattern matching.

Reply:
So funny how "what's easier to implement" leaks into the protocol. And then varies by language. No wonder the IETF is notorious for being contentious.

Reply:
To be fair to the Phoenix folks, liveview isn't really a protocol—it's all written internally in a single repo/project, and there's no spec.

Reply:
point taken